Sources/AsyncHTTPClient/RequestBag.swift

```diff
@@ -154,8 +154,11 @@ final class RequestBag<Delegate: HTTPClientResponseDelegate> {
             return self.task.eventLoop.makeFailedFuture(error)
 
         case .write(let part, let writer, let future):
-            writer.writeRequestBodyPart(part, request: self)
-            self.delegate.didSendRequestPart(task: self.task, part)
+            let promise = self.task.eventLoop.makePromise(of: Void.self)
+            promise.futureResult.whenSuccess {
+                self.delegate.didSendRequestPart(task: self.task, part)
```

Comment:
Should we report an error to the delegate in the failure case?

Reply:
Was thinking about that as well. Open to input here. @weissi For your backpressure implementation would this be valuable?

Reply:
We should make sure the error delegate is only called once.